Library/Homebrew/cmd/reinstall.rb

```diff
@@ -126,6 +131,98 @@ def run
         unless formulae.empty?
           Install.perform_preinstall_checks_once
 
+          ask_input = lambda {
+            ohai "Do you want to proceed with the installation? [Y/y/yes/N/n]"
```

Comment:
Seems like a lot of duplicated code here, would be good to extract into shared method(s) in `Library/Homebrew/install.rb`